docs/spec-files/general_messaging_constructs.md

```diff
@@ -24,19 +24,18 @@ Attachments are contained within a list in the `attachments` header.
 
 Each attachment is contained within the following structure.
 
-- `id`: Identifies attached content within the scope of a given message. Recommended on appended attachment descriptors. Possible but generally unused on embedded attachment descriptors. Never required if no references to the attachment exist; if omitted, then there is no way to refer to the attachment later in the thread, in error messages, and so forth. Because `id` is used to compose URIs, it is recommended that this name be brief and avoid spaces and other characters that require URI escaping.
+- `id`: [optional but recommended] Identifies attached content within the scope of a given message, so it can be referenced. For example, in a message documenting items for sale on an auction website, there might be a field named `front_view` that contains the value `#attachment1`; this would reference an attachment to the message with `id` equal to `attachment1`. Recommended on appended attachment descriptors. If omitted, then there is no way to refer to the attachment later in the thread, in error messages, and so forth. Because `id` is used to compose URIs, this value should be brief and MUST consist entirely of [unreserved URI characters](https://datatracker.ietf.org/doc/html/rfc3986/#section-2.3) – meaning that it is not necessary to [percent encode](https://en.wikipedia.org/wiki/Percent-encoding) the value to incorporate it in a URI.
```

Comment:
```suggestion
- `id`: [recommended] Identifies attached content within the scope of a given message, so it can be referenced. For example, in a message documenting items for sale on an auction website, there might be a field named `front_view` that contains the value `#attachment1`; this would reference an attachment to the message with `id` equal to `attachment1`. Recommended on appended attachment descriptors. If omitted, then there is no way to refer to the attachment later in the thread, in error messages, and so forth. Because `id` is used to compose URIs, this value should be brief and MUST consist entirely of [unreserved URI characters](https://datatracker.ietf.org/doc/html/rfc3986/#section-2.3) – meaning that it is not necessary to [percent encode](https://en.wikipedia.org/wiki/Percent-encoding) the value to incorporate it in a URI.
```

Would it make sense to use urn:didcommv2:id:<message-thread-id>#<attachmentId> so that we can reference attachments across messages? Id prefer to default to using just #<attachmentId> for simplicity, but thinking this would be beneficial in some optimization cases.

Reply:
yes always fully qualified

Reply:
There is a whole proposal about DIDComm message hyperlinking. I tried to put it in the DIDComm spec but was told to take it out as unnecessary. But the requirements as written are intended to support linking across messages. You don't need to make the references fully qualified *inside* a message (this isn't JSON-LD where we are trying to canonicalize and sign); you just need to know how to convert a relative reference to absolute.

Reply:
I prefer not to do the conversion so as long as I can use fully qualified which I assume is still valid I am fine with relative.